### PR TITLE
Openai embedding endpoint

### PR DESCRIPTION
**Pull request by magdyksaleh**
Adding an OpenAI compatible embedding endpoint. 

Also adding a batch embed functionality

### CODE REVIEW

**Review by tgaddair — APPROVED** [review]
Sweet! LGTM!

File: router/src/server.rs [hunk 1]
```diff
@@ -1643,6 +1645,69 @@ async fn embed(
...
+)]
+#[instrument(skip_all)]
+#[axum::debug_handler]
+async fn compat_embed(
+    infer: Extension<Infer>,
+    Json(req): Json<CompatEmbedRequest>,
+) -> Result<Json<CompatEmbedResponse>, (StatusCode, Json<ErrorResponse>)> {
+    metrics::increment_counter!("lorax_request_count");
+    tracing::debug!("Input: {}", req.input);
+    if let StringOrVec::Vec(inputs) = req.input {
```
Comment (arnavgarg1) [inline router/src/server.rs:1667]: OOC what's the difference between string or vec as inputs? Is it the vec just a list of strings and why we have a .map call?

Comment (magdyksaleh) [reply router/src/server.rs:1667]: yeah but we need to handle the case that the input is a string or a list of strings. Rust is a bit weird like that. So you need a special type for that. https://platform.openai.com/docs/api-reference/embeddings/create#embeddings-create-input

Comment (arnavgarg1) [reply router/src/server.rs:1667]: Very interesting, good to know!

**Pull request merged**